pkg/probe/http/request_test.go

```diff
@@ -38,3 +44,53 @@ func TestFormatURL(t *testing.T) {
 		}
 	}
 }
+
+func Test_userAgent(t *testing.T) {
+	tests := []struct {
+		name    string
+		purpose string
+		want    string
+	}{
+		{
+			name:    "test useragent output",
+			purpose: "agent",
+			want:    "kube-agent/.",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := userAgent(tt.purpose); got != tt.want {
+				t.Errorf("userAgent() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_v1HeaderToHTTPHeader(t *testing.T) {
+	tests := []struct {
+		name       string
+		headerList []v1.HTTPHeader
+		want       http.Header
+	}{
+		{
+			name: "test1",
```

Comment:
can you add a case with an empty input to assert the behavior?

Reply:
Updated.f

Reply:
I mean, another test case, 

```
{
			name: "empty input",

}
```

also, use a more meaningful name for this test , `test1` is not very informative, `common headers` per example, ...